data-prepper-plugins/ml-inference-processor/src/main/java/org/opensearch/dataprepper/plugins/ml_inference/processor/common/SageMakerBatchJobCreator.java

```diff
@@ -50,55 +63,149 @@ public SageMakerBatchJobCreator(final MLProcessorConfig mlProcessorConfig, final
         this.awsCredentialsSupplier = awsCredentialsSupplier;
         this.s3Client = createS3Client(mlProcessorConfig, awsCredentialsSupplier);
         this.dateTimeFormatter = DateTimeFormatter.ofPattern("yyyyMMddHHmmss");
+        this.MAX_BATCH_SIZE = mlProcessorConfig.getMaxBatchSize();
+
+        // Initialize batch processing scheduler
+        scheduler.scheduleAtFixedRate(
```

Comment:
Actually, you don't need this. processor's `doExecute()` is always invoked even when there is no traffic. You can use that.

Reply:
Great point! I just noticed each worker runs every 3 seconds. Will make a change to use it which makes things easier.